src/scan.ts

```diff
@@ -26,21 +26,21 @@ import { getProperties } from './properties';
 import { initializeAxios } from './request';
 import { downloadScannerCli, runScannerCli, tryLocalSonarScannerExecutable } from './scanner-cli';
 import { fetchScannerEngine, runScannerEngine } from './scanner-engine';
-import { ScanOptions, ScannerProperty } from './types';
+import { ScanOptions, ScannerProperty, CliArgs } from './types';
 
-export async function scan(scanOptions: ScanOptions, cliArgs?: string[]) {
+export async function scan(scanOptions: ScanOptions, cliArgs?: CliArgs) {
   try {
     await runScan(scanOptions, cliArgs);
   } catch (error: any) {
     log(LogLevel.ERROR, `An error occurred: ${error?.message ?? error}`);
   }
 }
 
-async function runScan(scanOptions: ScanOptions, cliArgs?: string[]) {
+async function runScan(scanOptions: ScanOptions, cliArgs?: CliArgs) {
   const startTimestampMs = Date.now();
   const properties = getProperties(scanOptions, startTimestampMs, cliArgs);
 
-  if (properties[ScannerProperty.SonarVerbose] === 'true') {
+  if (properties[ScannerProperty.SonarVerbose] === 'true' || cliArgs?.debug) {
```

Comment:
NP (feel free to ignore)

Couldn't we keep the current code here
```suggestion
  if (properties[ScannerProperty.SonarVerbose] === 'true') {
```

and modify `getCommandLineProperties(cliArgs)` to add `properties[ScannerProperty.SonarVerbose] = 'true'` when `cliArgs.debug`?

The functional result will be the same, but semantically, it would make the property-related logic fully encapsulated in `properties.ts`